ruby/example_code/iam/scenario_create_user_assume_role.rb

```diff
@@ -0,0 +1,295 @@
+  def create_s3_resource(credentials)
+    Aws::S3::Resource.new(client: Aws::S3::Client.new(credentials: credentials))
+  end
+
+  # Lists buckets for the account, using the specified Amazon S3 resource. The resource
+  # uses credentials with limited access, so it may not be able to list buckets.
+  #
+  # @param s3_resource [Aws::S3::Resource] An Amazon S3 resource.
+  def list_buckets(s3_resource)
+    count = 10
+    s3_resource.buckets.each do |bucket|
+      puts "\t#{bucket.name}"
+      count -= 1
+      break if count.zero?
+    end
+  rescue Aws::Errors::ServiceError => e
+    if e.code == 'AccessDenied'
+      puts('Attempt to list buckets with no permissions: AccessDenied.')
+    else
+      puts("Couldn't list buckets for the account. Here's why: ")
+      puts("\t#{e.code}: #{e.message}")
+      raise
+    end
+  end
+
+  # snippet-start:[ruby.example_code.sts.AssumeRole]
+  # Creates an AWS STS client with specified credentials. This is separated into a
+  # factory function so that it can be mocked for unit testing.
+  #
+  # @param key_id [String] The ID of the access key used by the STS client.
+  # @param key_secret [String] The secret part of the access key used by the STS client.
+  def create_sts_client(key_id, key_secret)
+    Aws::STS::Client.new(access_key_id: key_id, secret_access_key: key_secret)
+  end
+
+  # Get temporary credentials that can be used to assume a role.
+  #
+  # @param role_arn [String] The ARN of the role that is assumed when these credentials
+  #                          are used.
+  # @param sts_client [AWS::STS::Client] An AWS STS client.
+  # @return [Aws::AssumeRoleCredentials] The credentials that can be used to assume the role.
+  def assume_role(role_arn, sts_client)
+    credentials = Aws::AssumeRoleCredentials.new(
+      client: sts_client,
+      role_arn: role_arn,
+      role_session_name: 'create-use-assume-role-scenario'
+    )
+    puts("Assumed role '#{role_arn}', got temporary credentials.")
+    credentials
+  end
+  # snippet-end:[ruby.example_code.sts.AssumeRole]
+
+  # snippet-start:[ruby.example_code.iam.DeleteRole]
+  # Deletes a role. If the role has policies attached, they are detached and
+  # deleted before the role is deleted.
+  #
+  # @param role [Aws::IAM::Role] The role to delete.
+  def delete_role(role)
+    role.attached_policies.each do |policy|
+      name = policy.policy_name
+      policy.detach_role(role_name: role.name)
+      policy.delete
+      puts("Deleted policy #{name}.")
+    end
+    name = role.name
+    role.delete
+    puts("Deleted role #{name}.")
+  rescue Aws::Errors::ServiceError => e
+    puts("Couldn't detach policies and delete role #{role.name}. Here's why:")
+    puts("\t#{e.code}: #{e.message}")
+    raise
+  end
+  # snippet-end:[ruby.example_code.iam.DeleteRole]
+
+  # snippet-start:[ruby.example_code.iam.DeleteUser]
+  # Deletes a user. If the user has inline policies or access keys, the are deleted
```

Comment:
change "the are deleted" to "they are deleted"

Reply:
Fixed